File: modules/expo-move-to-back/android/src/main/java/expo/modules/movetoback/ExpoMoveToBackModule.kt
package expo.modules.movetoback

import android.content.Context
import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

class ExpoMoveToBackModule : Module() {
    override fun definition() = ModuleDefinition {
        Name("ExpoMoveToBack")

        Function("moveToBack") {
            // moveTaskToBack is an Activity/UI operation and must run on the
            // main thread; this Function is invoked on the JS thread. Dispatch
            // to the UI thread (fire-and-forget — the call itself is trivial).
            // Safe-call (not `?: return@Function`) so the body's return shape
            // stays Unit? — the bare labelled return doesn't satisfy the Expo
            // `Function` DSL's expected return type and fails to compile.
            appContext.currentActivity?.let { activity ->
                activity.runOnUiThread { activity.moveTaskToBack(true) }
            }
        }

        Function("setSkipPreviousBehavior") { behavior: String ->
            val context = appContext.reactContext ?: appContext.currentActivity?.applicationContext
            val sp = context?.getSharedPreferences("substreamer_settings", Context.MODE_PRIVATE)
            sp?.edit()?.putString("skip_previous_behavior", behavior)?.apply()
        }
    }
}

File: modules/expo-move-to-back/src/ExpoMoveToBackModule.ts
import { requireNativeModule } from 'expo-modules-core';

interface ExpoMoveToBackInterface {
  moveToBack: () => void;
  setSkipPreviousBehavior?: (behavior: string) => void;
}

let module: ExpoMoveToBackInterface;

try {
  module = requireNativeModule('ExpoMoveToBack');
} catch {
  console.warn(
    '[expo-move-to-back] Native module not found. ' +
      'Run `npx expo run:ios` or `npx expo run:android` to rebuild with the native module.'
  );

  module = {
    moveToBack: () => {},
    setSkipPreviousBehavior: () => {},
  };
}

export default module;
